homeassistant/components/sensor/mysensors.py

```diff
@@ -129,37 +129,35 @@ def state(self):
-                self.gateway.const.SetReq.V_UNIT_PREFIX in self._values:
-            prefix = self._values[self.gateway.const.SetReq.V_UNIT_PREFIX]
-        if self.value_type == self.gateway.const.SetReq.V_TEMP:
-            return TEMP_CELCIUS  # HA will convert to degrees F if needed
-        elif self.value_type == self.gateway.const.SetReq.V_HUM or \
-                self.value_type == self.gateway.const.SetReq.V_DIMMER or \
-                float(self.gateway.version) >= 1.5 and \
-                self.value_type == self.gateway.const.SetReq.V_PERCENTAGE or \
-                self.value_type == self.gateway.const.SetReq.V_LIGHT_LEVEL:
-            return '%'
-        elif self.value_type == self.gateway.const.SetReq.V_WEIGHT:
-            return '{}g'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_DISTANCE:
-            return '{}m'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_IMPEDANCE:
-            return '{}ohm'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_WATT:
-            return '{}W'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_KWH:
-            return '{}kWh'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_FLOW:
-            return '{}m'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_VOLUME:
-            return '{}m3'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_VOLTAGE:
-            return '{}V'.format(prefix)
-        elif self.value_type == self.gateway.const.SetReq.V_CURRENT:
-            return '{}A'.format(prefix)
+        # HA will convert to degrees F if needed
+        unit_map = {
+            self.gateway.const.SetReq.V_TEMP: TEMP_CELCIUS,
+            self.gateway.const.SetReq.V_HUM: '%',
+            self.gateway.const.SetReq.V_DIMMER: '%',
+            self.gateway.const.SetReq.V_LIGHT_LEVEL: '%',
+            self.gateway.const.SetReq.V_WEIGHT: 'kg',
+            self.gateway.const.SetReq.V_DISTANCE: 'm',
+            self.gateway.const.SetReq.V_IMPEDANCE: 'ohm',
+            self.gateway.const.SetReq.V_WATT: 'W',
+            self.gateway.const.SetReq.V_KWH: 'kWh',
+            self.gateway.const.SetReq.V_FLOW: 'm',
+            self.gateway.const.SetReq.V_VOLUME: 'm3',
+            self.gateway.const.SetReq.V_VOLTAGE: 'V',
+            self.gateway.const.SetReq.V_CURRENT: 'A',
+        }
+        unit_map_v15 = {
+            self.gateway.const.SetReq.V_PERCENTAGE: '%',
+        }
+        custom_unit = ''
+        if float(self.gateway.version) >= 1.5:
+            if self.gateway.const.SetReq.V_UNIT_PREFIX in self._values:
+                custom_unit = self._values[
+                    self.gateway.const.SetReq.V_UNIT_PREFIX]
+            unit_map.update(unit_map_v15)
+        if custom_unit:
+            return custom_unit
+        if self.value_type in unit_map:
```

Comment:
A `dict` has a `get(key, default=None)` method to retrieve keys without raising exceptions. You could rewrite these three lines as:
```python
return unit_map.get(self.gateway.const.SetReq(self.value_type))
```